syft/pkg/cataloger/golang/licenses.go

```diff
@@ -23,23 +23,36 @@ import (
 	"github.com/anchore/syft/internal/licenses"
 	"github.com/anchore/syft/internal/log"
 	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/internal/cache"
 	"github.com/anchore/syft/syft/internal/fileresolver"
 	"github.com/anchore/syft/syft/pkg"
 )
 
 type goLicenses struct {
-	catalogerName         string
-	opts                  CatalogerConfig
-	localModCacheResolver file.WritableResolver
-	lowerLicenseFileNames *strset.Set
+	catalogerName          string
+	opts                   CatalogerConfig
+	localModCacheResolver  file.Resolver
+	remoteModCacheResolver file.WritableResolver
+	lowerLicenseFileNames  *strset.Set
 }
 
 func newGoLicenses(catalogerName string, opts CatalogerConfig) goLicenses {
+	var localModCacheResolver file.Resolver
+	if opts.SearchLocalModCacheLicenses {
+		localModCacheResolver = fileresolver.NewFromUnindexedDirectory(opts.LocalModCacheDir)
+	}
+
+	var remoteModCacheResolver file.WritableResolver
+	if opts.SearchRemoteLicenses {
+		remoteModCacheResolver = cache.Resolver("go-mod")
```

Comment:
I think we'd want a way to naturally version these. Specifically, being able to know when some existing cache is compatible with the current code. I feel like all callers should specify this explicitly:
```
cache.Resolver("go-mod", "v1")
```
over
```
cache.Resolver("go-mod-v1")
```
since it is easy for callers to not know when to coordinate breaking changes.